File: app/schemas/visualization.py
from pydantic import BaseModel

from app.schemas.llm import LLMProviderName


class VisualizeRequest(BaseModel):
    connection_id: int
    table_name: str
    prompt: str
    llm_provider: LLMProviderName | None = None
    llm_model: str | None = None
    row_limit: int | None = None


class VisualizeResponse(BaseModel):
    history_id: int
    execution_status: str
    image_base64: str | None = None
    generated_code: str | None = None
    error_message: str | None = None
    duration_ms: int

File: frontend/src/types/visualization.ts
import type { LLMProviderName } from "./llm";

export interface VisualizeRequest {
  connection_id: number;
  table_name: string;
  prompt: string;
  llm_provider?: LLMProviderName;
  llm_model?: string;
  row_limit?: number;
}

export type ExecutionStatus = "success" | "error" | "timeout";

export interface VisualizeResponse {
  history_id: number;
  execution_status: ExecutionStatus;
  image_base64: string | null;
  generated_code: string | null;
  error_message: string | null;
  duration_ms: number;
}
